syft/format/cyclonedxxml/encoder_test.go

```diff
@@ -21,6 +24,34 @@ func getEncoder(t testing.TB) sbom.FormatEncoder {
 	return enc
 }
 
+func TestCondensedOutput(t *testing.T) {
+	enc, err := NewFormatEncoderWithConfig(EncoderConfig{
+		Version: cyclonedxutil.DefaultVersion,
+		Compact: true,
+	})
+	require.NoError(t, err)
+
+	dir := t.TempDir()
+	s := testutil.DirectoryInput(t, dir)
+
+	var buffer bytes.Buffer
+	err = enc.Encode(&buffer, s)
+	require.NoError(t, err)
+
+	actual := buffer.String()
+	lines := strings.Split(actual, "\n")
+	require.NotEmpty(t, lines)
+	whitespace := regexp.MustCompile(`^\s+`)
+	for _, line := range lines {
+		if len(line) == 0 {
+			continue
+		}
+
+		// require a non-whitespace character (tab, space, etc) as the first character of the line
+		require.False(t, whitespace.Match([]byte(line)), "line should not start with whitespace: %q", line)
```

Comment:
Does compact XML still have newlines in it?

Reply:
yes